src/frequenz/dispatch/_actor_dispatcher.py

```diff
@@ -32,6 +32,9 @@ class DispatchInfo:
     options: dict[str, Any]
     """Additional options."""
 
+    duration: timedelta | None
+    """The duration of the dispatch."""
+
```

Comment:
Just suggestion: `duration` can be in `options` dict.
However I don't know why you need this so feel free to ignore this comment.